cudax/include/cuda/experimental/__stf/utility/unique_id.cuh

```diff
@@ -33,34 +33,47 @@ namespace reserved
 /* This defines an object with a unique identifier. This object is non
  * copyable, but moving it transfers the unique id to the destination object.
  */
-template <typename C, C...>
+template <typename C>
 class unique_id
 {
 public:
   unique_id() = default;
-  constexpr unique_id(unique_id&& other) noexcept
+  unique_id(unique_id&& other) noexcept
       : _value(::std::exchange(other._value, -1))
-  {}
-  constexpr unique_id(const int val) noexcept
+  {
+    assert(_value >= 0 && "Reading a dead unique_id is not allowed.");
+  }
+  unique_id(const int val) noexcept
       : _value(val)
-  {}
-  constexpr operator int() const noexcept
   {
+    assert(val >= 0 && "A unique_id must contain a non-negative value.");
```

Comment:
Is there a reason you are removing the `constexpr from that functions?`

Reply:
I think `constexpr` is superfluous in this class, but I'll add it back for the sake of "constexpr all the things".